### PR TITLE
Add GetReg and UpdateReg methods

### PR DESCRIPTION
**Pull request by sgomes**
@crhym3 PTAL

### CODE REVIEW

File: acme.go [hunk 1]
```diff
@@ -86,33 +86,26 @@ func (c *Config) CertSource() CertSource {
...
+	nonce, err := fetchNonce(client, url)
 	if err != nil {
 		return err
 	}
 
-	// prepare new-reg request
+	// prepare registration request
 	reg := struct {
 		Resource  string   `json:"resource"`
 		Contact   []string `json:"contact"`
```
Comment (crhym3) [inline acme.go]: should the contact be tagged with `omitempty`?

File: acme.go [hunk 2]
```diff
@@ -130,10 +123,18 @@ func Register(client *http.Client, config *Config) error {
...
 		return responseError(res)
 	}
 
+	var resp struct {
+		Contact []string `json:"contact"`
+	}
+	if err := json.NewDecoder(res.Body).Decode(&resp); err != nil {
+		return fmt.Errorf("Decode: %v", err)
+	}
+	config.Contact = resp.Contact
```
Comment (crhym3) [inline acme.go]: move this beneath `// update config with the response` comment

File: acme.go (hunk 1 above)
Comment (sgomes) [inline acme.go]: Yep. Done.

File: acme.go (hunk 2 above)
Comment (sgomes) [inline acme.go]: Done.

**Comment by crhym3** [comment]
LGTM
I'll start working on the cmd line now. Nice work.

**Comment by sgomes** [comment]
Thanks!

**Pull request merged**